src/main/java/org/truffleruby/core/rope/TruffleRopesNodes.java

```diff
@@ -135,7 +135,9 @@ private static String getStructure(ConcatRope rope) {
         }
 
         private static String getStructure(SubstringRope rope) {
-            return getStructure(rope.getChild()) + "[" + rope.getByteOffset() + ", " + rope.characterLength() + "]";
+            final Rope child = rope.getChild();
+            final int characterOffset = RopeOperations.strLength(child.getEncoding(), child.getBytes(), 0, rope.getByteOffset());
```

Comment:
It's debug code, I'd rather not depend on the logic in StringByteCharacterIndexNode.
I could move it to SubstringRope.getCharacterIndex() but I'm not sure it's a good idea.